src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonMigrator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsProjectJsonMigrator))]
+    class VsProjectJsonMigrator : IVsProjectJsonMigrator
+    {
+        private IVsSolutionManager _solutionManager;
+        private PumpingJTF _pumpingJtf;
+
+        [ImportingConstructor]
+        public VsProjectJsonMigrator()
+        {
+            _solutionManager = ServiceLocator.GetInstance<IVsSolutionManager>();
+            _pumpingJtf = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory.Context);
+        }
+        public IVsProjectJsonMigrateResult MigrateProjectToPackageRef(string projectUniqueName)
+        {            
+            return _pumpingJtf.Run(async delegate
+            {
+                return await MigrateProjectToPackageRefAsync(projectUniqueName);
+            });
+            
+        }
+
+        private async Task<IVsProjectJsonMigrateResult> MigrateProjectToPackageRefAsync(string projectUniqueName)
+        {
+            var vsSolMngr = _solutionManager as VSSolutionManager;
```

Comment:
It looks like this should be cast in the constructor and verified there. This extra var and the name is somewhat confusing.